05.ls/ls4.rb

```diff
@@ -0,0 +1,134 @@
+# frozen_string_literal: true
+
+require 'optparse'
+require 'etc'
+
+class Ls
+  COLUMN = 3
+  def initialize
+    opt = ARGV.getopts('a', 'r', 'l')
+    @directories = if opt['a']
+                     Dir.glob('*', File::FNM_DOTMATCH)
+                   elsif opt['r']
+                     Dir.glob('*').reverse
+                   else
+                     Dir.glob('*')
+                   end
+  end
+
+  def divide_directories
+    if (@directories.size % COLUMN).zero?
+      divide_directories = @directories.each_slice(@directories.size / COLUMN).to_a
+    else
+      divide_directories = @directories.each_slice(@directories.size / COLUMN + 1).to_a
+      until divide_directories.first.size == divide_directories.last.size
+        # 後述するtransposeメソッドを使うために、要素数が不足している配列に、nilを代入する。
+        divide_directories.last << nil
+      end
+    end
+    divide_directories
+  end
+
+  def sort_directories
+    sorted_directories = divide_directories.transpose
+    max_length = @directories.max_by(&:length).length
+    sorted_directories.each do |x|
+      x.each do |y|
+        print format("%-#{max_length + 1}s", y)
+      end
+      puts
+    end
+  end
+
+  def run_option_l
+    total = []
+
+    @directories.map! do |file|
+      total << get_blocks(file)
+      { type: get_type(file), permission: get_permission(file), hardlink: get_hardlink(file), user: get_user(file), group: get_group(file), size: get_size(file),
+        time: get_time(file), filename: file }
```

Comment:
こういう感じで書いてあげた方がみやすいかもしれません。好みの問題でもありますが参考までに。

```suggestion
      { 
        type: get_type(file),
        permission: get_permission(file),
        hardlink: get_hardlink(file),
        user: get_user(file),
        group: get_group(file),
        size: get_size(file),
        time: get_time(file),
        filename: file
      }
```

Reply:
上記の記法に変更しました。